### DISCUSSION TITLE
Go back in e2e tests with Detox pressing headerback button

### DISCUSSION THREAD

**Discussion by Andarius** [react-navigation/react-navigation#11335] [Q&A]
Hi,

I'm trying to understand how to go back pressing the default header back button on Android / iOS using Detox. (I'm also using react-native-screens if it can help).
The button is visible:

![image](https://user-images.githubusercontent.com/5070712/232603122-05e92d3e-cbf0-4405-b736-3f472e5b8944.png)

I tried `element(by.id('header-back')).tap()` but it does not seems to work.
 
 I don't want to use `device.goBack()` as I want to test that the back button is visible and works.
 Any ideas ?

**Comment by traviswimer** [comment]
Why is this closed? Did you find a solution?

**Comment by Andarius** [comment]
Yes, I ended up using `element(by.label('Navigate up'))`. I'm not that happy with this solution and would prefer a `testID` but it works.
For reference, I also asked here for a better way to do this: https://github.com/wix/Detox/discussions/4022

**Reply by traviswimer** [reply]
Is `"Navigate up"` a default value added by `react-native-screens`?

What ended up working for me was `element(by.label("NameOfParentRoute"))`, which is the default label set by `react-navigation`. Although, this only works on iOS and I decided to just use `device.pressBack()` on Android.

It also looks like you can use the [`headerBackTitle` option](https://reactnavigation.org/docs/native-stack-navigator#headerbacktitle) which will also be set as the accessibility label, but it didn't work for me with `headerBackTitleVisible: false`. `headerBackTitle` also only works on iOS, so that wont work for Android either.

It seems odd to me that the API doesn't offer a way to at least set the `accessibilityLabel`, which would work on all platforms.

**Reply by Andarius** [reply]
I think it's the default Android value in ActionBar https://developer.android.com/reference/android/app/ActionBar#setHomeActionContentDescription(java.lang.CharSequence)

**Reply by julianD77** [reply]
This works for us in our App

```
export const pressBack = async () => {
    if (device.getPlatform() === 'android') {
        await device.pressBack();
    } else {
        await element(by.traits(['button']))
            .atIndex(0)
            .tap();
    }
};
```
